tests/components/metoffice/test_weather.py

```diff
@@ -327,3 +366,212 @@ async def test_two_weather_sites_running(
     assert weather.attributes.get("forecast")[2]["temperature"] == 11
     assert weather.attributes.get("forecast")[2]["wind_speed"] == 11.27
     assert weather.attributes.get("forecast")[2]["wind_bearing"] == "ESE"
+
+
+@pytest.mark.freeze_time(datetime.datetime(2020, 4, 25, 12, tzinfo=datetime.UTC))
+async def test_new_config_entry(hass: HomeAssistant, wavertree_data) -> None:
```

Comment:
Should this test and the test `test_legacy_config_entry` really be in this file? Maybe we should move them to `test_init`

Reply:
I think it should be here, it tests `homeassistant/components/metoffice/weather.py::async_setup_entry`